components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -2954,6 +2954,82 @@ export function createWalletApi () {
         providesTags: (res, er, arg) => [
           { type: 'SolanaEstimatedFees', id: arg.txId }
         ]
+      }),
+      getEthTokenDecimals: query<
+        number,
+        Pick<BraveWallet.BlockchainToken, 'chainId' | 'contractAddress'>
+      >({
+        queryFn: async (arg, { endpoint }, extraOptions, baseQuery) => {
+          try {
+            const { jsonRpcService } = baseQuery(undefined).data
+            const { errorMessage, decimals } =
+              await jsonRpcService.getEthTokenDecimals(
+                arg.contractAddress,
+                arg.chainId
+              )
+
+            if (!decimals) {
+              throw new Error(errorMessage)
+            }
+
+            return {
+              data: Number(decimals)
+            }
+          } catch (error) {
+            return handleEndpointError(
+              endpoint,
+              `Unable to fetch token decimals for ${arg.contractAddress}`,
+              error
+            )
+          }
+        },
+        providesTags: (res, err, arg) =>
+          err
+            ? ['UNKNOWN_ERROR']
+            : [
+                {
+                  type: 'EthTokenDecimals',
+                  id: [arg.chainId, arg.contractAddress].join('-')
+                }
+              ]
+      }),
+      getEthTokenSymbol: query<
```

Comment:
I see you get decimals, symbol. You might want to consider fetching the name as well

Reply:
Oh I didn't know `name` was also part of the ERC20 interface. Good to know! We don't need the token name for this particular case (it doesn't get displayed), but I'll add it another PR.

Reply:
yep, it is, although is optional (although most contracts have them).

<img width="1152" alt="image" src="https://github.com/brave/brave-core/assets/2352112/43bd28bf-bdb8-4981-a4b8-3252c5b82c5c">

Not sure if your provider accounts for it, but there's some contracts that implement these methods but return bytes instead of the string. One that comes to mind is `SAI`, so I would try this out. They have different ABI for the same method, technically they don't respect the standard, but most dapps handle this case.

Reply:
Thanks for the info. Created an issue to track this: https://github.com/brave/brave-browser/issues/33162